zrcs/zrc-3.md

```diff
@@ -0,0 +1,431 @@
+(* @param recipient:  Address of the recipient whose balance is increased. *)
+(* @param amount:     Amount of tokens to be sent.                         *)
+transition Send(from: ByStr20, recipient: ByStr20, amount: Uint128)
+```
+
+|        | Name        | Type      | Description                                          |
+| ------ | ----------- | --------- | ---------------------------------------------------- |
+| @param | `from`      | `ByStr20` | Address of the sender whose balance is decreased.    |
+| @param | `recipient` | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`    | `Uint128` | Amount of tokens to be sent.                         |
+
+|           | Name          | Description                | Event Parameters                                                                                                                                                                                                                   |
+| --------- | ------------- | -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `SendSuccess` | Sending is successful.     | `from`: `ByStr20`, `recipient`: `ByStr20`, and `amount`: `Uint128`.                             |
+| eventName | `Error`       | Sending is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not the contract owner. |
+
+#### 4. SendCheck
+
+```ocaml
+(* @dev: Moves amount tokens from sender to recipient via metatransaction. *)
+(* @param tokenOwner: Address of the sender whose balance is decreased.    *)
+(* @param recipient:  Address of the recipient whose balance is increased. *)
+(* @param amount:     Amount of tokens to be sent.                         *)
+(* @param checkHash:  teller-supplied hash of the metatransaction check.   *)
+(* @param checkSig:   Signature of checkHash - supplied by the tokenOwer.  *)
+(* @param tip:        Amount of tokens tokenOwner allocated for the teller *)
+(* @param nonce:      The account-specific nonce to use for transaction    *)
+transition SendCheck(tokenOwner: ByStr20, recipient: ByStr20, amount: Uint128, checkHash:ByStr, checkSig: ByStr64, tip: Uint128, nonce: Uint128)
+```
+
+|        | Name         | Type      | Description                                          |
+| ------ | ------------ | --------- | ---------------------------------------------------- |
+| @param | `tokenOwner` | `ByStr20` | Address of the sender whose balance is decreased.    |
+| @param | `recipient`  | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`     | `Uint128` | Amount of tokens to be sent.                         |
+| @param | `checkHash` | `ByStrX` | The concatenated hashes of the metacheck fields to validate|
```

Comment:
`ByStrX` isn't a valid Scilla type. `X` should be concrete or omitted. i.e., for example, `ByStr100` or `ByStr`